src/makeCalculator.js

```diff
@@ -4,7 +4,48 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculate = {
+    result: 0,
+
+    add(number) {
+      this.result += number;
+    },
+
+    subtract(number) {
+      this.result -= number;
+    },
+
+    multiply(number) {
+      this.result *= number;
+    },
+
+    divide(number) {
+      if (number === 0) {
+        console.log('Cannot divide by zero');
+        return;
+      }
+
+      this.result /= number;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(callback, number) {
+      const isCalculatorMethod = [this.add, this.subtract, this.multiply, this.divide].includes(callback);
+
+      if (isCalculatorMethod && typeof callback === 'function') {
```

Comment:
The check for `typeof callback === 'function'` is redundant here since you're already checking if the callback is one of the calculator methods in the previous line. You can safely remove this check.